src/modules/kvs/kvs.c

```diff
@@ -812,7 +829,13 @@ static int kvstxn_cache_cb (kvstxn_t *kt, struct cache_entry *entry, void *data)
         kvstxn_cleanup_dirty_cache_entry (kt, entry);
         return -1;
     }
+
+    /* must be true, otherwise we didn't insert entry in cache */
```

Comment:
Oh, maybe `cache_entry_create()` should have a blobref argument and perform the strdup so this is always true?  Then since the key is in the entry, `cache_insert()` wouldn't need a `ref` argument?

Reply:
That's a good idea.  I had assumed there were some locations in the code where a cache entry was created before data was available for it, but that's not the case.  And now that I think about it, that situation doesn't even seem to make sense.  This would be a good change.